### PR TITLE
Add 128-bit trace id generation and propagation via _dd.p.tid

### PR DESCRIPTION
**Pull request by bwoebi**
### Description

Adds a new `DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED=false` config option.

### Readiness checklist
- [x] (only for Members) Changelog has been added to the release document.
- [x] Tests added for this feature/bug.

### Reviewer checklist
- [x] Appropriate labels assigned.
- [x] Milestone is set.
- [x] Changelog has been added to the release document. For community contributors the reviewer is in charge of this task.

### CODE REVIEW

File: ext/span.c [hunk 1]
```diff
@@ -140,6 +140,11 @@ void ddtrace_open_span(ddtrace_span_data *span) {
     // All open spans hold a ref to their stack
     ZVAL_OBJ_COPY(&span->property_stack, &stack->std);
 
+    span->duration_start = _get_nanoseconds(USE_MONOTONIC_CLOCK);
+    // Start time is nanoseconds from unix epoch
+    // @see https://docs.datadoghq.com/api/?lang=python#send-traces
+    span->start = _get_nanoseconds(USE_REALTIME_CLOCK);
```
Comment (morrisonlevi) [inline ext/span.c:146]: This seems unrelated to the other changes. Are we expecting these to be set elsewhere, and this is being defensive, or was this a bug?

**Review by morrisonlevi — APPROVED** [review]

File: ext/span.c (hunk 1 above)
Comment (morrisonlevi) [reply ext/span.c:146]: I see it's related, but I need to read the docs.

Comment (bwoebi) [reply ext/span.c:146]: Yeah, using the start time afterwards for the high bits.

File: ext/ddtrace.h [hunk 2]
```diff
@@ -75,7 +75,13 @@ typedef struct {
 
 typedef struct {
     uint64_t low;
-    uint64_t high;
+    union {
+        uint64_t high;
+        struct {
+            uint32_t padding; // zeroes
+            uint32_t time;
+        };
```
Comment (morrisonlevi) [inline ext/ddtrace.h:83]: Based on your commit comment, this is endian specific, yes?

File: ext/span.c [hunk 3]
```diff
@@ -163,7 +163,7 @@ void ddtrace_open_span(ddtrace_span_data *span) {
 set_trace_id_from_span_id:
         span->trace_id = (ddtrace_trace_id){
             .low = span->span_id,
-            .high = get_DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED() ? span->start : 0,
+            .time = get_DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED() ? span->start / 1000000000L : 0,
```
Comment (morrisonlevi) [inline ext/span.c:166]: I think you want `UINT64_C` instead of `L`:

```suggestion
            .time = get_DD_TRACE_128_BIT_TRACEID_GENERATION_ENABLED() ? span->start / UINT64_C(1000000000) : 0,
```

I'm not feeling well today, but this is going from nanoseconds down to seconds, I think? Just checking that's the intention? I'd add a comment about the unit in `ddtrace_trace_id.time`'s definition.

**Review by morrisonlevi — APPROVED** [review]
Approved, with some suggestions. Are we fine to merge this, though? Is the behavior solidified?

File: ext/ddtrace.h (hunk 2 above)
Comment (bwoebi) [reply ext/ddtrace.h:83]: Yes it does, I'll just do it (though we only really target little endian).

File: ext/span.c (hunk 3 above)
Comment (bwoebi) [reply ext/span.c:166]: I did just copy the L from somewhere else and it had the expected value :-D But UINT64_C is more correct, let me do that.

**Comment by bwoebi** [comment]
Yes, we're fine to merge that now :-)
